### PR TITLE
Feat(scoped variables): variable list

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
This PR brings changes to introduce UI Layout for users will be able to view list of saved variables along with names and description.

Screenshot 2023-08-22 at 4 49 33 PM

Dep: https://github.com/devtron-labs/dashboard/pull/1224
Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/scopedVariables/styles.scss [hunk 1]
```diff
@@ -405,3 +405,55 @@
     line-height: 20px;
     margin: 0px;
 }
+
```
Comment (shivani170) [inline src/components/scopedVariables/styles.scss:408]: Please reduce the css

File: src/components/scopedVariables/SavedVariables.tsx [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
 const SavedVariablesView = ({ scopedVariables, setScopedVariables }: SavedVariablesViewI) => {
     const [showDropdown, setShowDropdown] = useState<boolean>(false)
     const [currentView, setCurrentView] = useState<FileView>(FileView.YAML)
+    const [variablesList, setVariablesList] = useState<VariableListItemI[]>(null)
     const dropdownRef = useRef(null)
 
     const { status, progress, fileData, abortRead, readFile } = useFileReader()
 
+    useEffect(() => {
+        if (status?.status == null) {
```
Comment (shivani170) [inline src/components/scopedVariables/SavedVariables.tsx:28]: or Can we use  !status?.status

File: src/components/scopedVariables/SavedVariables.tsx [hunk 3]
```diff
@@ -1,20 +1,41 @@
...
+    VARIABLES_TEMPLATE,
+    SAMPLE_DATA,
+    TABLE_LIST_HEADINGS,
+} from './constants'
 import { ReactComponent as ICFileDownload } from '../../assets/icons/ic-file-download.svg'
 
 const SavedVariablesView = ({ scopedVariables, setScopedVariables }: SavedVariablesViewI) => {
     const [showDropdown, setShowDropdown] = useState<boolean>(false)
     const [currentView, setCurrentView] = useState<FileView>(FileView.YAML)
+    const [variablesList, setVariablesList] = useState<VariableListItemI[]>(null)
```
Comment (shivani170) [inline src/components/scopedVariables/SavedVariables.tsx:22]: Initialize it with empty array

File: src/components/scopedVariables/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { TableItemI, TableListI } from './types'
+
+export const TableList = ({ children, width, headings }: TableListI) => {
```
Comment (shivani170) [inline src/components/scopedVariables/Table.tsx:4]: Avoid using two exports in a single component

File: src/components/scopedVariables/SavedVariables.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/SavedVariables.tsx:28]: It is a boolean so the status false would mean error state

File: src/components/scopedVariables/Table.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    return (
+        <div className={`scoped-variables-list-container`}>
```
Comment (shivani170) [inline src/components/scopedVariables/Table.tsx:6]: No need of interpolation can directly use css

Comment (AbhishekA1509) [reply src/components/scopedVariables/Table.tsx]: Done

File: src/components/scopedVariables/SavedVariables.tsx [hunk 6]
```diff
@@ -1,20 +1,41 @@
...
     const dropdownRef = useRef(null)
 
     const { status, progress, fileData, abortRead, readFile } = useFileReader()
 
+    useEffect(() => {
+        if (status?.status == null) {
+            const variables = SAMPLE_DATA?.Variables?.map((item) => {
```
[the remark this answers is not in the document]
Comment (AbhishekA1509) [reply src/components/scopedVariables/SavedVariables.tsx]: Done

File: src/components/scopedVariables/SavedVariables.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/SavedVariables.tsx]: Done

File: src/components/scopedVariables/SavedVariables.tsx [hunk 7]
```diff
@@ -101,10 +122,18 @@ const SavedVariablesView = ({ scopedVariables, setScopedVariables }: SavedVariab
...
-                )}
-            </div>
+                </div>
+            ) : (
+                <TableList width={['200px', '429px']} headings={TABLE_LIST_HEADINGS}>
+                    {variablesList?.map((item) => (
+                        <TableItem
+                            key={item.name}
+                            columnsData={[item.name, item.description]}
+                            width={['200px', '429px']}
```
Comment (shivani170) [inline src/components/scopedVariables/SavedVariables.tsx:132]: Have to avoid hardcoded width. Can pass style to the component either

Comment (AbhishekA1509) [reply src/components/scopedVariables/SavedVariables.tsx]: Fixed

**Pull request closed without merging**